embeddings/hyperparameter_search/flair_configspace.py

```diff
@@ -67,6 +69,14 @@ def _parse_model_trainer_parameters(
         )
         return parameters, task_train_kwargs
 
+    @classmethod
+    def _parse_yaml(cls, path: T_path) -> Dict[str, Any]:
+        config = read_yaml(path)
+        variables = {"embedding_name": config.pop("embedding_name")}
```

Comment:
there's not stop condition on empty config somewhere so again why we're removing elements from original dict instead leaving them as they are?

Reply:
Okay I'll answer just here because it concerns all of the .pop() method comments. The reason for that is because at the end of the method we check for unmapped parameters. At the end of the method we call cls._check_unmapped_parameters(config) which do just that - check if we mapped all of the data from the **parameters** dictionary. If not then it may indicate some error on the user's side.

Reply:
That's what I figured after reading code further and because method is private not exposed to lib users publicly that's really up to us developers what happens inside but if public library would change content of my dict I'd be really suprised

Reply:
Then usually alternative approach would be not to remove keys and then in separately dedicated method let's say validate_if_some_param_was_ommited we'd compare diff of flattened collections and point which ones is missing - but again I understand that that's not a way of doing things here :)

Reply:
ok I see your point here and I think it would be valid if the user actually passed a dictionary for the HPS but the user passes the path to YAML file so the library does not really change the content of the user's object (dict in this case). We had this method ._check_for_unmapped_parameters() for parsing the config (e.g. in .parse_parameters() method) so it seemed handy to use it here to reuse the code we already wrote and not unnecessarily increase the number of methods in the library

is it okay to leave it as it is?

Reply:
Ok I see - to me it seems that we can leave it as it is because that's not core of this PR and we'll add followup to review use of pop vs get and other similar behaviours in codebase (or we'll wait for user feedback / question and know where to look :))